frontend/src/store/bucketStore.ts

```diff
@@ -30,5 +32,75 @@ export const useBucketStore = defineStore('bucket', () => {
     return bucket;
   }
 
-  return { loading, load, getBucketInfo, buckets };
+  async function getBucketPermissions(bucketId: string) {
+    try {
+      loading.value = true;
+
+      const searchPerms = (await bucketService.searchForPermissions({ bucketId })).data;
+
+      // TODO: Feed a comma separated list instead of searching individual users once COMS accepts that
+      const uniqueIds = [...new Set(searchPerms.map((x: any) => x.userId))]; //.join(',');
+      const uniqueUsers: any = [];
+
+      await Promise.all(
+        uniqueIds.map(async (x: any) => {
+          const userResponse = await userService.searchForUsers({ userId: x });
+          uniqueUsers.push(userResponse.data[0]);
+        })
+      );
+
+      const userPermissions: UserPermission[] = [];
+      uniqueUsers.forEach((user: any) => {
+        userPermissions.push({
+          userId: user.userId,
+          fullName: user.fullName,
+          create: searchPerms.find((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.CREATE) !== undefined,
+          read: searchPerms.find((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.READ) !== undefined,
+          update: searchPerms.find((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.UPDATE) !== undefined,
+          delete: searchPerms.find((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.DELETE) !== undefined,
+          manage: searchPerms.find((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.MANAGE) !== undefined,
+        });
+      });
+
+      permissions.value = userPermissions;
+    } finally {
+      loading.value = false;
+    }
+  }
+
+  async function addBucketPermission(bucketId: string, userId: string, permCode: string) {
+    try {
+      loading.value = true;
+
+      await bucketService.addPermissions(bucketId, [{ userId, permCode }]);
+    } finally {
+      loading.value = false;
+    }
```

Comment:
These try/finally loading toggles look like a great place to implement a higher order function wrapper.